chore(build): add sourcemaps

Thanood opened the pull request:
Adds sourcemaps to `build-system` task. :smile: 
Inspired by Vegar who was missing sourcemaps in his v-grid repo.

Thanood committed:
chore(build): add sourcemaps

JeroenVinke commented:
one issue I see with this is that it makes the files in the dist folder larger, so the payload plugin users need to download is larger. Can we make it so the sourcemap is only generated for the `devbuild` folder (the development equivalent of `dist`)?

Thanood commented:
Sure, that's the way it is done in skeleton-bridge and materialize-bridge (probably kendo, too).
`gulp serve` would also need to serve from there, right?

Edit: That's actually wrong, it's done in a seperate `dev` task in skeleton-bridge.

Thanood committed:
chrore(build): restrict sourcemap generation to development code

Thanood committed:
chore(deps): add ^ modifier to sample dependencies

Thanood commented:
Changed the sourcemap generation to only occur if not gulping `build-release`.
Also updated jspm dependencies of the sample to use `^` modifier (what's that called btw.?)..
